src/decorator-annotator.ts

```diff
@@ -159,9 +160,10 @@ export class DecoratorClassVisitor {
     if (valueName.getSourceFile() === this.rewriter.file) {
       return valueName;
     }
+    const typeStr = typeNode.getText();
     for (let i = 0; i < this.importedNames.length; i++) {
       const {name, declarationNames} = this.importedNames[i];
-      if (declarationNames.some(d => d === valueName)) {
+      if (typeStr.startsWith(name.text) && declarationNames.some(d => d === valueName)) {
```

Comment:
yes, this is about generics. changed to compare the first identifier in the type instead (using `===`)